lib/screens/forgot_password/forgot_password_repository.dart

```diff
@@ -56,3 +60,109 @@ class FakeForgotPasswordRepository extends ForgotPasswordRepository {
     }
   }
 }
+
+class APIForgotPasswordRepository extends ForgotPasswordRepository {
+  final String classTag = "APIForgotPasswordRepository";
+
+  ///Send [Email] [OTP] [Passoword]  Stage 3/3
+  @override
+  Future<void> changePassword(String email, String otp, String password) async {
+    final String tag = classTag + "changePassword";
+    http.Response response;
+    try {
+      response = await sl.get<http.Client>().post(
+        S.postChangePasswordUrl,
+        body: <String, dynamic>{S.emailKey: email, S.otpKey: otp, S.passwordKey: password},
+      );
+    } catch (e) {
+      Log.e(tag: tag, message: "NetworkError:" + e.toString());
+      throw NetworkException();
+    }
+
+    if (response.statusCode == 202) {
+      try {
+        return;
+      } catch (e) {
+        Log.e(tag: tag, message: "Error while decoding response json to get token: $e");
+        throw UnknownException();
+      }
```

Comment:
🔴 
don't copy code mindlessly. This try catch block makes no sense The tried block will never throw an exception, so what are you trying to catch? Correct it here an in the below functions too.

Reply:
Corrected as per FakeRepo (would have preferred bool to check at various instances but since cubit is already there would follow that)